qualtran/bloqs/swap_network/cswap_approx_test.py

```diff
@@ -53,8 +54,15 @@ def get_t_count_and_clifford(bc: Dict[Bloq, int]) -> Tuple[int, int]:
 
 
 @pytest.mark.parametrize("n", [*range(1, 6)])
-def test_t_complexity(n):
-    cq_testing.assert_decompose_is_consistent_with_t_complexity(CSwap(n))
+def test_t_complexity_cswap(n):
+    # Note: if you use the cirq decompose protocol, clifford counts won't
+    # match because cirq.FREDKIN has 14 cliffords per two-bit-cswap.
+    cswap = CSwap(n)
+    assert t_complexity(cswap) == t_complexity(cswap.decompose_bloq())
```

Comment:
@mpharrigan This is happening because `TwoBitCSwap` has both `as_cirq_op` and `decompose_from_registers` specified.  Because `TwoBitCSwap` is no longer an atomic gate; we should delete the `as_cirq_op` implementation from it. Once that is done, `assert_decompose_is_consistent_with_t_complexity` will pass here. 

I think it would be surprising for any users using the "compilation to cirq" strategy to get a circuit with a different T-count as what they expected using the bloq analysis. Note that the circuit structure in cirq style compilation can still be preserved because decomposing a `CSwap` now gives `BloqAsCirqGate(TwoBitCSwap())` instead of a `cirq.FREDKIN`; which is fine I think and works for all protocols. 

I tried making the change locally and it results in failure of only two tests (+ 2 unrelated failures due to https://github.com/quantumlib/Qualtran/pull/826) ; both of which should be updated / removed because we no longer want to have a 1:1 correspondence between `TwoBitCSwap` and `cirq.FREDKIN` 
```
FAILED qualtran/bloqs/basic_gates/swap_test.py::test_two_bit_cswap - assert cirq.FrozenCi...2)),\n    ),\n]) == cirq.FrozenCi...2)),\n    ),\n])
FAILED qualtran/bloqs/basic_gates/swap_test.py::test_cswap_cirq_decomp - AssertionError: Circuit's text diagram differs from the desired diagram.
```

What do you think?
```suggestion
def test_t_complexity_cswap(n):
    cq_testing.assert_decompose_is_consistent_with_t_complexity(CSwap(n))
```

Reply:
done